charts/netdata/templates/deployment.yaml

```diff
@@ -159,3 +160,156 @@ spec:
             claimName: {{ template "netdata.name" . }}-parent-alarms
         {{- end }}
       dnsPolicy: {{ .Values.parent.dnsPolicy }}
+
+---
+
+{{- if .Values.k8sState.enabled -}}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ template "netdata.name" . }}-k8s-state
+  labels:
+    app: {{ template "netdata.name" . }}
+    chart: {{ template "netdata.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+    role: k8sState
+spec:
+  selector:
+    matchLabels:
+      app: {{ template "netdata.name" . }}
+      release: {{ .Release.Name }}
+      role: k8sState
+  template:
+    metadata:
+      annotations:
+      {{- if .Values.k8sState.podAnnotationAppArmor.enabled }}
+        container.apparmor.security.beta.kubernetes.io/{{ .Chart.Name }}: unconfined
+      {{- end }}
+        checksum/config: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha256sum }}
+{{- with .Values.k8sState.podAnnotations }}
+{{ toYaml . | trim | indent 8 }}
+{{- end }}
+      labels:
+        app: {{ template "netdata.name" . }}
+        release: {{ .Release.Name }}
+        role: k8sState
+{{- with .Values.k8sState.podLabels }}
+{{ toYaml . | trim | indent 8 }}
+{{- end }}
+    spec:
+      securityContext:
+        fsGroup: 201
+      serviceAccountName: {{ .Values.serviceAccount.name }}
```

Comment:
service account should have create set to `true`

Reply:
Sorry, can't get it. Can you reword, please?

Reply:
Ah, I see. Then I guess we could make it a requirement for the whole thing - 0 reasons to create k8sState instance if either rbac.create or serviceAccount.create is not true.

Reply:
> service account should have create set to true

In theory, there could be a service account created manually/not from our helmchart - I guess that is the reason we have no if/else around it for [parent](https://github.com/netdata/helmchart/blob/53499b3e29cea279cb1e9585db1fd43e38a9bb19/charts/netdata/templates/deployment.yaml#L36)/[children](https://github.com/netdata/helmchart/blob/53499b3e29cea279cb1e9585db1fd43e38a9bb19/charts/netdata/templates/daemonset.yaml#L38-L39). If that is the case - we don't depend on `serviceAccount.create` here - a user can just set the name to "" if they want to use the default account.

Reply:
ok, so if `serviceAccount.create` is set to true I would leave `serviceAccount.name` empty  and add named templates:
```
{{- define "serviceAccountName.netdata" -}}
{{- if .Values.serviceAccount.create -}}
    {{ default (include "netadata.fullname" .) .Values.serviceAccount.name }}
{{- else -}}
    {{ default "default" .Values.serviceAccount.name }}
{{- end -}}
{{- end -}}
```
any options inside `serviceAccount` won't make helm fail, because of the `default`

this is rather a suggestion